R/lru.R

```diff
@@ -10,7 +18,13 @@ LRUcache_class <- R6::R6Class("LRUcache",
     },
     set = function(name, value) {
       stopifnot(is.character(name) && length(name) == 1)
-      if (length(private$data) == private$max_num && !(name %in% ls(private$data))) private$evict()
+      # Check if this value alone exceeds the cache size
+      size <- if(private$use_bytes) as.integer(as.character(pryr::object_size(value))) else 1
```

Comment:
Could you move this stuff to the private interface?
make all the checks in the `private$save`